### DISCUSSION TITLE
Doesn't seem to be working WITH SONGS? Keeps duplicating transcripts on several timestamps

### DISCUSSION THREAD

**Discussion by AIhasArrived** [openai/whisper#1833] [Q&A]
Hello,
I tried whisper with some song and Got this:
![image](https://github.com/openai/whisper/assets/148225825/698f8bfe-57c4-494e-a80f-67a29bce5512)
I tried large-v3 and "ro" language.
This line "Kimenikin ajtoumeleibe" only comes later, and several words were not registered before it.
Sometimes words and sentences are indeed repeated int this song but not like this,

Any way to improve this?

**Comment by ryanheise** [comment]
Whisper tends to get confused when there is no voice, such as when there are periods of music without any vocals. You can use Demucs to filter out the backing music first so that you're left just with the vocals, and then use VAD (e.g. Silaro VAD) to filter out the silent bits so that Whisper only analyses the parts with actual speech to be transcribed. [stable-ts](https://github.com/jianfch/stable-ts) is one tool that can do both of these for you, but there are many other tools that also at least do VAD.

**Comment by AIhasArrived** [comment]
You meant this vad thing (https://github.com/snakers4/silero-vad) ?
My question is:
1) But once the silences are removed, would not WHISPER get the wrong timestamps (relative to the original video)? Any solution for that? As in saving which silences was removed at what time so  you can compute  the right timestamp relative to the original video by adding or removing the value of duration of silences? 

2) Wow I did not know there were other tools to expand on whisper such as stable-ts, do you know of others aswell? I would like to explore them:)
Thanks

**Reply by ryanheise** [reply]
1. Tools such as stable-ts and others will also do post processing to re-insert the silent gaps into the final timestamps, so the outputted timestamps should align with the original audio.
2. See @EtienneAb3d 's comment below for another suggestion, as well as others you can find by searching this discussion board for "hallucinations"

While there are several tools that can do this, I think for the case of music you may find that Demucs can be helpful since it is specifically designed to separate the music from the vocals, so that you can feed the clean vocals into Whisper.

**Reply by AIhasArrived** [reply]
Yeah I actually tried demucs aswell, and did not get what I liked, I might hav got the wrogn language so I removed the langage setting,
Can I send you the sample of Music by email so you can try it yourself and see what's not working?

**Comment by EtienneAb3d** [comment]
Try large-v2!

I'm also currently preparing a new version of [WhisperHallu](https://github.com/EtienneAb3d/WhisperHallu) with a bunch of new features specially dedicated to music/song processing.
;-)

**Reply by AIhasArrived** [reply]
Very cool stuff, I am a bit.. overwhelmed, to be honest, there are so many good things to try, it seems

**Reply by EtienneAb3d** [reply]
To process songs, I think the best procedure is the following:

1. process the file with [WhisperHallu](https://github.com/EtienneAb3d/WhisperHallu) to get 2 files: a proper text (vocal extraction+noise/silence removal, etc) without timestamps and accurate timestamps (possibly with hallucinations).
2. process these to files with [WhisperTimeSync](https://github.com/EtienneAb3d/WhisperTimeSync) to send the proper timestamps at the right places over the proper text.

Knowing a new enriched version of WhisperHallu is in preparation...
;-)

**Reply by AIhasArrived** [reply]
I will make sure to come back to you and ask you plenty of questions If I am unable to to undestand somethign deal? ;)
I will be using it either soon or in 2 weeks (depending on some personal projects) Hope I will find there then.

**Reply by dgoryeo** [reply]
@EtienneAb3d great to hear that a new version of WhisperHallu is under the work. I saw this: [100% avoid Whisper's hallucinations](https://github.com/openai/whisper/discussions/679#discussioncomment-7686577) discussion the other day and I thought you might find it relevant / interesting --if you haven't already seen it.

**Reply by EtienneAb3d** [reply]
@dgoryeo 
As far as I understand it, it is filtering silence and noise parts while recording. This is a part of what WhisperHallu is doing on already-registered files using various tools.

**Reply by lengoc-tuyen** [reply]
@EtienneAb3d  @AIhasArrived,
Wow, I'm truly grateful to have found and read this chat!
I'm currently working on a similar project using Whisper to generate SRT files (I know, I'm a little late to the party!) I ran into all the same technical problems and this chat was incredibly helpful in guiding me through the fixes.
Since I'm just starting out in this field and feel quite lost , I wanted to ask: At this time, are there any newer or better tools/methods you would recommend to further improve the SRT accuracy and workflow?
I'm passionate about this project, so I'm happy to dedicate time and effort to try anything new you suggest.
Thank you again!
